src/use_case/login/LoginInteractor.java

```diff
@@ -0,0 +1,40 @@
+package use_case.login;
+
+import entity.User;
+import interface_adapter.menu.MenuViewModel;
+
+public class LoginInteractor implements LoginInputBoundary {
+    final LoginUserDataAccessInterface userDataAccessObject;
+    final LoginOutputBoundary loginPresenter;
+
+    public LoginInteractor(LoginUserDataAccessInterface userDataAccessInterface,
+                           LoginOutputBoundary loginOutputBoundary) {
+        this.userDataAccessObject = userDataAccessInterface;
+        this.loginPresenter = loginOutputBoundary;
+    }
+
+    @Override
+    public void execute(LoginInputData loginInputData) {
+        String username = loginInputData.getUsername();
+        String password = loginInputData.getPassword();
+        if (!userDataAccessObject.existsByName(username)) {
+            loginPresenter.prepareFailView(username + ": Account does not exist.");
+        } else {
+            String pwd = userDataAccessObject.get(username).getPassword();
+            if (!password.equals(pwd)) {
+                loginPresenter.prepareFailView("Incorrect password for " + username + ".");
```

Comment:
This fail view is not working for me, I'll open a bug report when this gets merged in.